mavros_extras/src/plugins/joy_control.cpp

```diff
@@ -0,0 +1,237 @@
+/**
+ * @brief JoyControl plugin
+ * @file joy_control.cpp
+ * @author Francois Chataigner <chataign@gmail.com>
+ *
+ * @addtogroup plugin
+ * @{
+ */
+/*
+ * Copyright 2016 Francois Chataigner.
+ *
+ * This file is part of the mavros package and subject to the license terms
+ * in the top-level LICENSE file of the mavros repository.
+ * https://github.com/mavlink/mavros/tree/master/LICENSE.md
+ */
+
+#include <mavros/mavros_plugin.h>
+#include <pluginlib/class_list_macros.h>
+
+#include <sensor_msgs/Joy.h>
+
+#include <mavros_msgs/SetMode.h>
+#include <mavros_msgs/CommandBool.h>
+#include <mavros_msgs/State.h>
+#include <mavros_msgs/SetMode.h>
+
+using namespace std;
+
+////////////////////////////////////////////////////////////////////////////////
+////////////////////////////////////////////////////////////////////////////////
+
+namespace mavplugin {
+struct Control
```

Comment:
Inside `class JoyControllPlugin` that ok, but here please add namespace or rename to `JoyControl`.